renderer/Image_load.cpp

```diff
@@ -552,11 +552,18 @@ void idImage::GenerateImage( const byte *pic, int width, int height,
 	GL_CheckErrors();
 	GL_SetDebugLabel( GL_TEXTURE, texnum, imgName );
 
-		//Routine test( &uploading );
-		auto start = Sys_Milliseconds();
+	//Routine test( &uploading );
+	auto start = Sys_Milliseconds();
+	if ( GLAD_GL_ARB_texture_storage && !generatorFunction ) {
+		int levels = 1 + idMath::Floor( idMath::ILog2( Max( scaled_width, scaled_height ) ) );
```

Comment:
If Floor necessary?
ILog2 is already integer-valued logarithm --- it only takes exponent from float representation.

Reply:
You're probably right. I took the formula from the glTexStorage manpage (https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glTexStorage2D.xhtml at the very bottom in the errors section), but yeah, ILog2 is already integer, so floor does nothing.

Reply:
Ok, I see.

Reply:
I removed Floor from the equation. As expected, does not appear to make any difference :)